### PR TITLE
fix(aws-eventbridge-sns): long sns topic names break eventbridge bindings

### LINKED ISSUE #1015: aws-eventbridge-sns fails validation for long topic names
When I use the `@aws-solutions-constructs/aws-eventbridge-sns` solution in conjunction with an SNS topic having a name longer than 64 characters, CloudFormation throws a validation error.

```
Error: The stack named ExampleLongStackName-Dev-Main failed to deploy: UPDATE_ROLLBACK_COMPLETE: 1 validation error detected: Value 'ExampleLongStackName-Dev-Main-LongExampleTopic2EDA8B02-DpDO44me9z' at 'targets.1.member.id' failed to satisfy constraint: Member must have length less than or equal to 64 (Service: AmazonCloudWatchEvents; Status Code: 400; Error Code: ValidationException; Request ID: 2695f1db-7bf0-470d-a89a-fee880abb30a; Proxy: null)
```

- The [CloudFormation guide for AWS::EventBridge::Rule Target](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-target.html#cfn-events-rule-target-id) lists `Id` as having a maximum of 64 characters.
- The [CloudFormation guide for AWS::SNS::Topic](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-topic.html#cfn-sns-topic-topicname) lists `TopicName` as having a maximum of 256 characters.

One option to fix the problem is to add an additional property to `EventbridgeToSnsProps` (e.g. named `eventTargetId`) that overrides the value supplied on this line:

https://github.com/awslabs/aws-solutions-constructs/blob/035ea20d42f2491403d363f937e05e50bb0da6ef/source/patterns/%40aws-solutions-constructs/aws-eventbridge-sns/lib/index.ts#L113

...if the `eventTargetId` is `undefined` it could just use the topic name as it's currently doing now.

### PR DESCRIPTION
**Pull request by biffgaut**
*Issue #, if available:*
Closes #1015 

*Description of changes:*
Explicitly set the physical name of the Event/Topic binding

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 3ff82380ec209a22a58b7ce60f917fa1fc78525a
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=68c1b470-3f79-446f-864d-ebaa1c0bfdfc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-eventbridge-sns/test/integ.second-long-stack-name.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import { Duration } from 'aws-cdk-lib';
+import { EventbridgeToSns, EventbridgeToSnsProps } from '../lib';
+import * as events from 'aws-cdk-lib/aws-events';
+import { App, Stack } from 'aws-cdk-lib';
+import { generateIntegStackName } from '@aws-solutions-constructs/core';
+
+const app = new App();
+
+const stackTwo = new Stack(app, generateIntegStackName(__filename) + 'ThisIsTheLongestNameForAStackItMustBeGreaterThanSixtyFourCharactersLongAndThisShouldJustAboutDoItNoItMustBeABitLessShort');
```
[the remark this answers is not in the document]
Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/aws-eventbridge-sns/test/integ.second-long-stack-name.ts:27]: No, cdk-integ will only manage 1 stack - if the file has 2 you need to have a specific comment instructing cdk-integ which stack to launch

File: source/patterns/@aws-solutions-constructs/aws-eventbridge-sns/test/eventbridge-sns-topic.test.ts [hunk 2]
```diff
@@ -141,7 +141,7 @@ test('check if the event rule has permission/policy in place in sns for it to be
   );
 });
 
-test('check events rule properties', () => {
+test.only('check events rule properties', () => {
```
[the remark this answers is not in the document]
Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/aws-eventbridge-sns/test/eventbridge-sns-topic.test.ts:144]: oops - I knew I was gonna forget that...

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: fab1f0992229ec3bd9baa7f21a2fdf2a3713d4e2
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8ea119ab-bf42-4f1d-a053-1d1777d35858%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by georgebearden — APPROVED** [review]
Looks good to me

**Pull request merged**